sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -258,8 +258,8 @@ def register(
         :type receiver: ~azure.servicebus.ServiceBusReceiver
         :param renewable: A locked entity that needs to be renewed.
         :type renewable: Union[~azure.servicebus.ServiceBusReceivedMessage, ~azure.servicebus.ServiceBusSession]
-        :param max_lock_renewal_duration: A time in seconds that the lock should be maintained for.
-          Default value is 300 (5 minutes).
+        :param max_lock_renewal_duration: A time in seconds that the lock should be maintained for. If set to None,
+         `max_lock_renewal_duration` value passed to constructor will be used. Default value is None.
```

Comment:
I updated the default value in the docstring, rather than the actual default value from None to 300. I think doing so might be a breaking change for the following scenario:
For example, if user passes in `max_lock_renewal_duration=500` and doesn't pass in anything in `register` here (so `max_lock_renewal_duration=None`), then `500` would be passed in on line 312. However, changing the default to 300 here means that the value passed in to line 312 will be 300, if the customer doesn't touch their code.

Reply:
You could probably just state something like:
```python
If specified, this value will override the default value specified at the constructor.
```

Reply:
yep, I agree that we should only change the docstring and your description about the behavior is precise :)

Reply:
We don't need to update the docs for the async auto-renewer?

Reply:
out of curiosity: is it that the api generator would collect information (default value 300) from docstring for api view?

Reply:
I thought there was only a shared auto renewer (since it's in the _common folder), but I might be mistaken?

Reply:
Travis confirmed that default can be parsed from docstring if not specified in the signature.